torch/_dynamo/guards.py

```diff
@@ -2877,10 +2877,19 @@ def install_guard(*guards, skip=0):
     """
     from torch._guards import TracingContext
 
-    collect_debug_stack = guards_log.isEnabledFor(
-        logging.DEBUG
-    ) or verbose_guards_log.isEnabledFor(logging.DEBUG)
-    add = TracingContext.get().guards_context.dynamo_guards.add
-    for guard in guards:
-        assert isinstance(guard, Guard)
-        add(guard, collect_debug_stack=collect_debug_stack, skip=skip + 1)
+    if not config.unsafe_skip_all_guards:
+        collect_debug_stack = guards_log.isEnabledFor(
+            logging.DEBUG
+        ) or verbose_guards_log.isEnabledFor(logging.DEBUG)
+        add = TracingContext.get().guards_context.dynamo_guards.add
+        for guard in guards:
+            assert isinstance(guard, Guard)
+            add(guard, collect_debug_stack=collect_debug_stack, skip=skip + 1)
```

Comment:
I tried locally to comment out this line, but I still get some guards:

```python
[rank0]:V0822 00:17:15.035000 626512 torch/_dynamo/guards.py:2704] [0/1] [__recompiles_verbose] Recompiling function forward in /home/youkaichao/vllm/vllm/worker/tpu_model_runner.py:599
[rank0]:V0822 00:17:15.035000 626512 torch/_dynamo/guards.py:2704] [0/1] [__recompiles_verbose]     triggered by the following guard failure(s):
[rank0]:V0822 00:17:15.035000 626512 torch/_dynamo/guards.py:2704] [0/1] [__recompiles_verbose]     guard 0 failures:
[rank0]:V0822 00:17:15.035000 626512 torch/_dynamo/guards.py:2704] [0/1] [__recompiles_verbose]     - 0/0: Eq(Mod(L['token_ids'].size()[1], 16), 0)                      # _dynamo/output_graph.py:455 in init_ambient_guards; 2 <= L['token_ids'].size()[1]                                 # _dynamo/output_graph.py:455 in init_ambient_guards; 2 <= L['attn_metadata'].slot_mapping.size()[1]                # _dynamo/output_graph.py:455 in init_ambient_guards
```

is it related to some 0/1 specialization? I vaguely remember some related terms from @ezyang 's podcast.

Reply:
Ah. this is not the only place we are inserting guards. I will have to sprinkle this flag at more places then.

Reply:
Try the latest commit.

Reply:
then I get error:

```python
[rank0]:   File "<eval_with_key>.1 from /home/youkaichao/miniconda3/envs/myenv/lib/python3.10/site-packages/torch/fx/experimental/proxy_tensor.py:910 in wrapped", line 9, in forward
[rank0]:   File "/home/youkaichao/miniconda3/envs/myenv/lib/python3.10/site-packages/torch/_ops.py", line 720, in __call__
[rank0]:     return self._op(*args, **kwargs)
[rank0]: RuntimeError: shape '[1]' is invalid for input of size 8
```

Reply:
This seems like - we should have recompiled. But skipping all the guards made Dynamo think that it can reuse the previously compiled code. 

I guess, this flag is not so useful for you. What you want is to have more control over what to guard and what not to guard.

Reply:
Converting this to WIP because I think we need more discussion on what kind of control is suitable for vLLM usecase.